python/pysrc/kaskada/_timestream.py

```diff
@@ -1009,6 +1009,51 @@ def cast(self, data_type: pa.DataType) -> Timestream:
         """
         return Timestream(self._ffi_expr.cast(data_type))
 
+    def seconds_since(self, time: Union[Timestream, Literal]) -> Timestream:
+        """
+        Create a Timestream containing the number of seconds since `time`.
+
+        Parameters
+        ----------
+        n : Union[Timestream, Literal]
+            The time to compute the seconds since.
+
+            This can be either a stream of timestamps or a datetime literal.
+
+        Returns
+        -------
+        Timestream
+            Timestream containing the number of seconds since `time`.
+        """
+        if isinstance(time, datetime):
+            session = self._ffi_expr.session()
+            nanos = Timestream._literal(time.timestamp() * 1e9, session=session)
+            nanos = Timestream.cast(nanos, pa.timestamp("ns", None))
+            return Timestream._call("seconds_between", nanos, self).cast(pa.int64())
+        else:
+            return Timestream._call("seconds_between", time, self).cast(pa.int64())
```

Comment:
Why does this cast back to integer vs. leaving it as `seconds` (the duration unit)?

Reply:
Changed back to using `duration('s')` and modified `_previous` as well to do the same. Casted explicitly in the tests.